collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2953,6 +2956,11 @@ public void onNumberPickerValueSelected(int widgetId, int value) {
         }
     }
 
+    @Override
+    public void onDateChanged(LocalDateTime date) {
+        ((ODKView) currentView).setBinaryData(date);
```

Comment:
Dangerous access of `currentView`. Can you wrap this in an `instanceof` block?

Reply:
Fixed using https://github.com/grzesiek2010/collect/commit/e979e292bf285f318b1c5c7150cbc96e0f01775a